### DISCUSSION TITLE
Bookmarks & History

### DISCUSSION THREAD

**Discussion by N3C2L** [madprops/grasshopper#81] [Ideas]
I am very grateful to you and do not want to put you under pressure.
I understand that what I am asking for is quite a bit, but perhaps some things can be implemented step by step. I don't want to overwhelm you; consider my thoughts as inspirations to possibly implement some aspects if you find them appealing.

I'm posting this here because it's a mix of issues and ideas.

1. Folders are not displayed in Grasshopper in the Bookmarks Section, but that could perhaps be implemented:
Maybe a dual view, above the bookmarked website items and below the folder box. If a folder is clicked, it's content is displayed above. The advantage of this dual view is that no folders need to be collapsed or expanded (as is the case with the addons "Sidebery" or "Bookmark Plus 2"). In the Folder box, you can double-click individual folders to navigate into other folders within them. When a search is performed, search results are displayed in item list and below in the folder box.
If you were to implement folders, other things would come handy such as selecting 5 tabs > bookmarking > auto popup to name a folder > prompting to delete the tab items. I know from the firefox store description that you don't want to touch deletion methods of bookmarks at all. But maybe some adding of bookmarks features like the folders or similar could be introduced?

2. I would like to use bookmark and history functions, but the lack of websites favicons & folders is holding me back.
In the addon "Sidebery" favicons are displayed in the History + Bookmark sections, and there is also the option to search by folders. (The search bar must be activated in Sidebery in the options). In the Bookmark Plus 2 addon, there is no history function, but here too, you can search by folders, and favicons are displayed.

3. Another issue:
The filterings in the Bookmark and History sections at the top in the Top button bar do not work. For example, tags or colors are not displayed when these filterings are activated. However, it would be possible to introduce other filterings in the History Section, such as month and day. And in the Bookmark Section, filterings like recently bookmarked could be introduced.

4. In the History Section, there might be the possibility to label individual items with a symbol if it is a closed tab. Alternatively, you could introduce a Closed Tabs Box below in the History Section.

These repositories could be useful:

https://github.com/aaFn/Bookmark-search-plus-2
https://github.com/mbnuqw/sidebery

**Comment by madprops** [comment]
About the favicons: Those extensions save the images to local storage, since the favicon is simply not supplied by firefox on history or bookmarks, which is a shame. Another approach is making network requests to a favicon server, but it's not something that makes me comfortable.

Made a check saving the favicons. In one load it saved over 700kb of favicons. That's going to fill the 5mb quickly unless the unlimited storage permission is used.

**Reply by N3C2L** [reply]
Thanks for adding the favicons!

**Reply by madprops** [reply]
I'm not adding favicons for now. I don't like the solutions .. but if I did it would use the google favicon url, which is basically a hack

**Comment by madprops** [comment]
Ok next version will use the favicon url.

**Comment by madprops** [comment]
Also, color etc filtering in history and bookmarks are there in the filter menu because items edited through domain rules affect those items too. If a history item is red because of a domain rule, it will appear in the red filter

**Reply by N3C2L** [reply]
Oh, thank you, I didn't know that. And thank you for using the google favicon url as a solution, I am looking forward to the next version.

**Comment by N3C2L** [comment]
There is a small bookmark section bug: right click on a bookmark > more > "bookmark" cmd is listed here.

Maybe add an option in the right click menu of a bookmark to rename the bookmarked title.

**Comment by madprops** [comment]
With bookmarks I try to do as less as possible to avoid messing with the important bookmarks of people accidentally. I rely on the browser interface for editing and removal, which is easy with the "star" icon on the url bar etc

**Comment by N3C2L** [comment]
Alright. Thanks for putting the domain rule in the right click menu of the bookmarks, it's useful there. 
And thanks for the fetching favicons feature.

**Comment by N3C2L** [comment]
Another thought:
You don't need to allow deleting bookmarks from the drive in the grasshopper interface. You can introduce a Trash folder and introduce "move to trash" cmds, where users can move bookmarks into. This trash folder and it's files can only be deleted by the native firefox bookmark interface and not in the grasshopper interface.

**Comment by N3C2L** [comment]
This ideas won't be implemented, as far as I can observe, so it can be closed.
